setup.py

```diff
@@ -117,6 +121,70 @@ def _have_openmp():
     omp_lib = []
 
 
+if not have_pep621_support:
+    metadata = dict(
+        name="kite",
+        use_scm_version=True,
+        description="Handle SAR displacement data towards pyrocko",
+        author="Marius P. Isken, Henriette Sudhaus;"
+        "BriDGes Emmily Noether-Programm (DFG)",
+        author_email="mi@gfz-potsdam.de",
+        long_description=open("README.md", "r").read(),
+        long_description_content_type="text/markdown",
+        license="GPLv3",
+        classifiers=[
+            "Intended Audience :: Science/Research",
+            "Topic :: Scientific/Engineering",
+            "Topic :: Scientific/Engineering :: Atmospheric Science",
+            "Topic :: Scientific/Engineering :: Image Recognition",
+            "Topic :: Scientific/Engineering :: Physics",
+            "Topic :: Scientific/Engineering :: Visualization",
+            "Programming Language :: Python :: 3.5",
+            "Programming Language :: C",
+            "Operating System :: POSIX",
+            "Operating System :: MacOS",
+        ],
+        keywords=["InSAR satellite radar earthquake optimization"],
+        python_requires=">=3.7",
+        setup_requires=["setuptools_scm[toml]>=5.0.0"],
+        install_requires=[
+            "numpy>=1.17.3",
+            "scipy>=1.6.0",
+            "PyQt5>=5.15.0",
+            "pyqtgraph>=0.11.0",
+            "pyrocko>=2022.06.10",
+            "utm>=0.7.0",
+            "geojson>=2.5.0",
+        ],
+        extras_require={
+            "gdal": ["gdal>=3.2.0"],
+            "development": ["flake8", "black", "pre-commit"],
+            "tests": ["pytest"],
+        },
+        packages=[
+            "kite",
+            "kite.util",
+            "kite.sources",
+            "kite.spool",
+            "kite.talpa",
+            "kite.talpa.sources",
+        ],
+        package_data={"kite": ["spool/res/*", "talpa/res/*"]},
+        entry_points={
+            "console_scripts": [
+                "spool = kite.spool.__main__:main",
+                "talpa = kite.talpa.__main__:main",
+                "stamps2kite = kite.util.stamps2kite:main",
+                "bbd2kite = kite.util.bbd2kite:main",
+            ]
+        },
+        ext_package="kite",
+    )
+
+else:
+    metadata = {}
+
+
```

Comment:
Why no update pip? There should be only one source of truth. The `pyproject.toml`. Don't work against pip.

Reply:
True  - *should* be. But updating pip would lead to a mess in a system-wide install scenario. Need to have a fallback as long as PEP621 is not fully supported by commodity distributions.

Reply:
Relaxed constrains on the dependencies to facilitate distribution packages, and a (preferred) user-space installation should do it imo.

Reply:
That is a very tight constraint: all downstream projects relying on kite will then also have to enforce users to use the same policy. The relaxed dependency constraints do not solve the installation problem. The problem is the old pip in the standard distributions and you don't want to mess around with that.

Reply:
Plus: there is no error message or any kind of feedback if you naively do a normal system-wide installation. No one will understand why kite doesn't work in that circumstance.

Reply:
> That is a very tight constraint: all downstream projects relying on kite will then also have to enforce users to use the same policy. The relaxed dependency constraints do not solve the installation problem. The problem is the old pip in the standard distributions and you don't want to mess around with that.

The user should upgrade pip in userspace to a recent version. An upgrade should even be encouraged!